src/content_sao.cpp

```diff
@@ -1208,6 +1216,36 @@ int PlayerSAO::punch(v3f dir,
 
 	if (!damage_handled) {
 		setHP(getHP() - hitparams.hp);
+
+		if (!isDead()) {
+			if (g_settings->getBool("enable_knockback")) {
+				int knockback_power = toolcap->knockback_power;
+				f32 knockback_x = fabs(puncher->getBasePosition().X - m_base_position.X);
+				f32 knockback_z = fabs(puncher->getBasePosition().Z - m_base_position.Z);
+				float total = knockback_x + knockback_z;
+				v3f direction;
+				static const f32 knockback_horizontal = g_settings->getFloat("knockback_horizontal");
+				static const f32 knockback_vertical = g_settings->getFloat("knockback_vertical");
+				static const f32 time_reset_knockback = g_settings->getFloat("knockback_default_duration");
+
+				if (knockback_x != 0) {
+					direction.X = -((((knockback_x / total) * knockback_horizontal) *
```

Comment:
I already manage it. I do division if knockbase_x and knockbase_y are different than 0